### PR TITLE
feat: transaction screen

### LINKED ISSUE #38: Create Transaction screen
## Figma reference
https://www.figma.com/file/uEckQY4EaQ5rXwyISg6d1g/Fuel-Wallet?node-id=290%3A15917

## URL Params
- `transactionId: String`

## Blocked by
- https://github.com/FuelLabs/fuels-wallet/issues/31
- https://github.com/FuelLabs/fuels-wallet/issues/32
- https://github.com/FuelLabs/fuels-wallet/issues/34
- https://github.com/FuelLabs/fuels-wallet/issues/35
- https://github.com/FuelLabs/fuels-wallet/issues/36
- https://github.com/FuelLabs/fuels-wallet/issues/37
- #303 
- #291 
- #290 
- #289 
- #288 
- #286 
- #284

### PR DESCRIPTION
**Pull request by LuizAsFight**
Closes #38

### CODE REVIEW

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Transaction/utils/index.tsx [hunk 1]
```diff
@@ -15,13 +16,13 @@ export function parseTransaction<T extends TxRequest | TxResponse>(tx: T) {
   }, {} as T);
 }
 
-export function getCoinInputsFromTx(tx?: TxRequest) {
-  return (tx?.inputs ?? []).filter(
+export function getCoinInputsFromTx(tx?: TxRequest | Transaction) {
```
[the remark this answers is not in the document]
Comment (LuizAsFight) [reply packages/app/src/systems/Transaction/utils/index.tsx:19]: it works for the `getCoinInputsFromTx`, but not for `getCoinOutputsFromTx`.

then I kept both in same way. we have this error:
image

File: packages/app/src/systems/Transaction/machines/transactionMachine.tsx [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      }),
+      assignGetTransactionError: assign({
+        error: (_) => TRANSACTION_ERRORS.NOT_FOUND,
+      }),
+      assignGetTransactionResultError: assign({
+        error: (_) => TRANSACTION_ERRORS.RECEIPTS_NOT_FOUND,
+      }),
+      clearError: assign({
+        error: (_) => TRANSACTION_ERRORS.RECEIPTS_NOT_FOUND,
+      }),
```
Comment (luizstacio) [inline packages/app/src/systems/Transaction/machines/transactionMachine.tsx:137]: I think `clearError` should return undefined or null

File: packages/app/src/systems/Transaction/machines/transactionMachine.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        };
+      }),
+      assignGetTransactionResult: assign((_, event) => {
+        const txResult = event.data as TransactionResult<any>;
+
+        return {
+          txResult,
+          transaction: txResult.transaction,
+          txStatus:
+            txResult.status.type === 'success'
```
Comment (luizstacio) [inline packages/app/src/systems/Transaction/machines/transactionMachine.tsx:155]: Do we have an enum for status type? I think `fuels` has.

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Comment by luizstacio** [comment]
1. When the transaction fee is too low, it doesn't show the amount. It should show 0.000000001. Example txId: 0x755fecd0059b835efbcd0e647590c38f87b95ca062ec978f28a085042c05e3a9
Screenshot 2022-12-08 at 00 44 00
2. Also, it's possible to show how much has been sent or received in any transaction is possible to cross the InputCoins with the Output of type `Change`. Ex. txId: 0x18617ccc580478214175c4daba11903df93a66a94aada773e80411ed06b6ade7
Screenshot 2022-12-08 at 00 58 31

File: packages/app/src/systems/Transaction/machines/transactionMachine.tsx (hunk 2 above)
Comment (LuizAsFight) [reply packages/app/src/systems/Transaction/machines/transactionMachine.tsx:137]: thanks! ctrlcctrlv feelings 😳

File: packages/app/src/systems/Transaction/machines/transactionMachine.tsx (hunk 3 above)
Comment (LuizAsFight) [reply packages/app/src/systems/Transaction/machines/transactionMachine.tsx:155]: it doesn't:

image

File: packages/app/src/systems/Transaction/hooks/useTransaction.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
+import type { Coin } from '@fuel-wallet/types';
```
Comment (luizstacio) [inline packages/app/src/systems/Transaction/hooks/useTransaction.tsx:1]: I think this film has room for more refactoring, creating more utility functions, and reducing the amount of `[0]` being used.

Comment (luizstacio) [reply packages/app/src/systems/Transaction/hooks/useTransaction.tsx:1]: Also, we could create a test for this hook once it has important logic, or by splitting in utilities create tests for the `utils/tx.tsx`.

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**